src/theory/datatypes/sygus_extension.cpp

```diff
@@ -1196,7 +1196,7 @@ void SygusExtension::registerSymBreakLemma( TypeNode tn, Node lem, unsigned sz,
         sca.d_search_terms[tn].find(d);
     if (itt != sca.d_search_terms[tn].end())
     {
-      for (const TNode& t : itt->second)
+      for (const Node& t : itt->second)
```

Comment:
Why not `(TNode t : itt->second)` here?

Reply:
So this would be the difference of copying a `Node` to a `TNode` versus holding a reference to a `Node`. I'm not sure it makes a difference, and I went with having `const T&` for a container that holds `T` as a generic pattern.

Reply:
Right, and I actually agree that with that rule of thumb. I was just wondering since we decided in one of the meetings to use `TNode` instead of `const Node&` as a rule of thumb (and I don't really agree with this rule of thumb).

Reply:
We said that for function arguments, mainly because it allows to be called with either Node (cheap copy to TNode) or TNode without a lot of overhead. In this case we actually know it's a Node, so it's a bit different.